tests/integration/tests/test_user_config.py

```diff
@@ -63,3 +66,85 @@ def test_user_config(instances: List[harness.Instance]):
         in result.stdout.decode()
     )
     instance.exec("rm /var/snap/k8s/common/args/k8s-env".split())
+
+
+@pytest.mark.node_count(1)
+@pytest.mark.bootstrap_config((config.MANIFESTS_DIR / "bootstrap-all.yaml").read_text())
+@pytest.mark.tags(tags.NIGHTLY)
+def test_no_unnecessary_helm_revisions(instances: List[harness.Instance]):
+    """Verifies that calling 'k8s set' with the same configuration multiple times
+    does not create new helm chart revisions.
+
+    Tests that:
+    - Multiple calls to 'k8s set' with the same values don't trigger unnecessary helm upgrades
+    - The ck-network helm chart revision count remains stable across repeated calls
+    """
+    instance = instances[0]
+
+    # Bootstrap the cluster
+    util.wait_until_k8s_ready(instance, [instance])
+
+    def get_helm_release_revision(chart_name: str, namespace: str) -> str:
+        """Get the revision number of a specific helm release."""
+        result = instance.exec(
+            [
+                "k8s",
+                "helm",
+                "list",
+                "-n",
+                namespace,
+                "-f",
+                chart_name,
+                "-o",
+                "json",
+            ],
+            capture_output=True,
+            text=True,
+        )
+
+        releases = json.loads(result.stdout)
+        if not releases:
+            return "0"  # Chart not found
+
+        return releases[0]["revision"]
+
+    # Get initial revision of ck-network chart
+    initial_revision = get_helm_release_revision("ck-network", "kube-system")
+    LOG.info("Initial ck-network chart revision: %s", initial_revision)
+
+    # Call 'k8s set' with the same configuration multiple times
+    for i in range(3):
+        LOG.info("Running k8s set command (iteration %d)", i + 1)
+        instance.exec(
+            "k8s set network.enabled=true gateway.enabled=true ingress.enabled=true".split()
```

Comment:
Any reason to not do this for all the features? If not I feel like it's best we test it all.